src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -225,6 +307,26 @@ def _smime_encode(
     return fp.getvalue()
 
 
+def _smime_enveloped_encode(data: bytes) -> bytes:
+    # This function works pretty hard to replicate what OpenSSL does
+    # precisely. For good and for ill.
```

Comment:
Nitpick: might be good to include a URL or code reference for OpenSSL's construction of the encoding here, just in case this ever needs to be re-evaluated 🙂

Reply:
This is actually a comment from the other encoder function that I used as a base for this one (`_smime_signed_encode`). 

 I think it doesn't make much sense here since we just add the headers, so I removed it